controller/src/controller/subsystems/instrument_comm.py

```diff
@@ -300,49 +309,112 @@ async def _catch_expired_command(self) -> None:
     async def _handle_comm_from_monitor(self) -> None:
         while True:
             comm_from_monitor = await self._from_monitor_queue.get()
-
             bytes_to_send = bytes(0)
             packet_type: int | None = None
 
-            match comm_from_monitor:
-                case {"command": "start_stim_checks", "well_indices": well_indices}:
-                    packet_type = SerialCommPacketTypes.STIM_IMPEDANCE_CHECK
-                    bytes_to_send = struct.pack(
-                        f"<{NUM_WELLS}?",
-                        *[
-                            STIM_MODULE_ID_TO_WELL_IDX[module_id] in well_indices
-                            for module_id in range(NUM_WELLS)
-                        ],
-                    )
-                case {"command": "set_stim_protocols", "stim_info": stim_info}:
-                    packet_type = SerialCommPacketTypes.SET_STIM_PROTOCOL
-                    bytes_to_send = convert_stim_dict_to_bytes(stim_info)
-                    if self._is_stimulating and not self._hardware_test_mode:
-                        raise InstrumentCommandAttemptError(
-                            "Cannot update stimulation protocols while stimulating"
+            is_offline_mode_comm = comm_from_monitor["command"] in (
+                "end_offline_mode",
+                "check_connection_status",
+            )
+
+            ignore_incoming_comm = not is_offline_mode_comm and self._system_in_offline_mode
+
+            if not ignore_incoming_comm:
```

Comment:
nitpick, I would prefer early exit here so we don't have to indent the match case statements which themselves already use a lot of indentation:
```
if ignore_incoming_comm:
   logger.info(
      f"Ignoring incoming command '{comm_from_monitor['command']}' in instrument comm while offline"
   )
   return
```